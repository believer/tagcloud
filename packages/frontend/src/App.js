import React from 'react'
import { Global, css } from '@emotion/core'
import styled from '@emotion/styled'
import Input from './components/Input'
import TagCloud from './components/TagCloud'
import { useDebounce } from '@iteam/hooks'

const globalStyles = css`
  * {
    box-sizing: border-box;
  }

  html,
  body {
    margin: 0;
  }
`

const Wrap = styled.div`
  display: grid;
  grid-template-rows: auto 1fr auto;
  height: 100vh;
`

const Header = styled.header`
  background-color: #24292e;
  color: #fff;
  padding: 20px;
  text-align: center;
`

const Main = styled.main`
  display: grid;
  grid-template-columns: 20px 1fr 20px;
  padding: 20px;

  @media (min-width: 1024px) {
    grid-template-columns: 1fr 600px 1fr;
  }
`

const MainInner = styled.div`
  grid-column: 2;
`

const Footer = styled.footer`
  background-color: #24292e;
  color: #fff;
  padding: 20px;
  text-align: center;
`

const App = () => {
  const [hashtag, setHashTag] = React.useState()
  const debouncedHashtag = useDebounce(hashtag, 300)

  return (
    <Wrap>
      <Global styles={globalStyles} />
      <Header>Tag Cloud by Rickard Laurin</Header>

      <Main>
        <MainInner>
          <Input onChange={setHashTag} value={hashtag} />
          <TagCloud hashtag={debouncedHashtag} />
        </MainInner>
      </Main>

      <Footer>
        email:
        <a href="mailto:rickard.laurin@gmail.com">rickard.laurin@gmail.com</a>
        github:
        <a href="https://github.com/believer">https://github.com/believer</a>
        website:
        <a href="https://rickardlaurin.se/">https://rickardlaurin.se/</a>
      </Footer>
    </Wrap>
  )
}

export default App
