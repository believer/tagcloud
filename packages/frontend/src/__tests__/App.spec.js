import React from 'react'
import { render } from 'react-testing-library'
import App from '../App'

test('should render', () => {
  const { getByText } = render(<App />)

  expect(getByText(/react/i)).toBeInTheDocument()
})
