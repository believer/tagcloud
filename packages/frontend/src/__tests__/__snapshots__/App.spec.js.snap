// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render words 1`] = `
<div>
  <div
    class="css-1dbsdhd"
  >
    <header
      class="css-1i14ur6"
    >
      Tag Cloud by Rickard Laurin
    </header>
    <main
      class="css-mvo6m"
    >
      <div
        class="css-lm4rz3"
      >
        <div
          class="css-1vwm2gx"
        >
          <label
            class="css-1yoaw6g"
            for="search"
          >
            Search by hashtag
          </label>
          <input
            class="css-bvcjre"
            id="search"
            type="text"
            value=""
          />
        </div>
        <div
          class="css-1797gyu"
        >
          <div
            class="css-1hk5syb"
            style="color: hsl(0, 0%, 0%); font-size: 48px;"
          >
            cookiemonster
          </div>
        </div>
      </div>
    </main>
    <footer
      class="css-1gpsrbn"
    >
      <ul>
        <li
          class="css-njh0gs"
        >
          E-mail:
           
          <a
            href="mailto:rickard.laurin@gmail.com"
          >
            rickard.laurin@gmail.com
          </a>
        </li>
        <li
          class="css-njh0gs"
        >
          GitHub:
           
          <a
            href="https://github.com/believer"
          >
            https://github.com/believer
          </a>
        </li>
        <li
          class="css-njh0gs"
        >
          Website:
           
          <a
            href="https://rickardlaurin.se/"
          >
            https://rickardlaurin.se/
          </a>
        </li>
      </ul>
    </footer>
  </div>
</div>
`;
